src/odemis/gui/util/img.py

```diff
@@ -1756,17 +1759,49 @@ def draw_legend_multi_streams(images, buffer_size, buffer_scale,
         if s:
             legend_ctx.show_text(s.name.value)
 
-        # If stream has tint, draw the colour in a little square next to the name
-        if (stream is None and
-            (isinstance(s, (acqstream.FluoStream, acqstream.StaticFluoStream)) or
-             hasattr(s, "tint") and tuple(s.tint.value) != (255, 255, 255)
-           )):
+        # Handle the stream colormap
+        if stream is None and hasattr(s, "tint"):
             tint = s.tint.value
-            legend_ctx.set_source_rgb(*conversion.rgb_to_frgb(tint))
-            legend_ctx.rectangle(legend_x_pos + cell_x_step - tint_box_size - small_font,
-                                 legend_y_pos - small_font,
-                                 tint_box_size, tint_box_size)
-            legend_ctx.fill()
+            
+            if isinstance(tint, colors.Colormap) or tint == FIT_TO_RGB and tint != (255, 255, 255):
+
+                # convert tint to a color map
+                if isinstance(tint, tuple):
+                    # swap r and b channels
+                    b, g, r = tuple(tint)
+                    tint = r, g, b
```

Comment:
I think that the reason you do this is because the internal cairo format is BGRA. So instead, you should convert the `gradient` (numpy array into a BGRA format), using `format_rgba_darray(gradient)`.